js_modules/dagster-ui/packages/ui-core/src/insights/LineChartWithComparison.tsx

```diff
@@ -0,0 +1,390 @@
+        if (typeof before !== 'number') {
+          return;
+        }
+
+        const after = before - timeSliceSeconds;
+
+        // Only open the dialog if data exists for the clicked index
+        // in the current period
+        if (metrics.currentPeriod.data[index] && openMetricDialog) {
+          openMetricDialog({
+            after,
+            before,
+            metric: metricName,
+            unit: unitType,
+          });
+        }
+      }
+    }
+  };
+
+  function getDisplayValueAndUnit(value: number) {
+    if (unitType === ReportingUnitType.TIME_MS) {
+      return formatDuration(value, {unit: 'milliseconds'})[0];
+    }
+    return {value, unit: unitTypeToLabel[unitType]};
+  }
+
+  const currentPeriodDisplayValueAndUnit = getDisplayValueAndUnit(
+    metrics.currentPeriod.aggregateValue ?? 0,
+  );
+  const prevPeriodDisplayValueAndUnit = getDisplayValueAndUnit(
+    metrics.prevPeriod.aggregateValue ?? 0,
+  );
+
+  return (
+    <div className={styles.chartContainer}>
+      <div className={styles.chartHeader}>
+        <Box flex={{direction: 'row', gap: 4, justifyContent: 'space-between'}}>
+          <BodyLarge>{metrics.title}</BodyLarge>
+          {loading ? <Spinner purpose="body-text" /> : null}
+        </Box>
+      </div>
+      <Box flex={{direction: 'column', justifyContent: 'space-between'}}>
+        <div className={styles.chartCount}>
+          {metrics.currentPeriod.aggregateValue
+            ? numberFormatterWithMaxFractionDigits(2).format(currentPeriodDisplayValueAndUnit.value)
+            : 0}
+          <Body color={Colors.textDefault()}>{currentPeriodDisplayValueAndUnit.unit}</Body>
+        </div>
+        <Box
+          className={styles.chartChange}
+          flex={{direction: 'row', gap: 4, justifyContent: 'space-between'}}
+        >
+          <Box flex={{direction: 'row', gap: 4, alignItems: 'center'}}>
+            {metrics.prevPeriod.aggregateValue
+              ? numberFormatterWithMaxFractionDigits(2).format(prevPeriodDisplayValueAndUnit.value)
+              : 0}
+            <span> previous period</span>
+          </Box>
+          <Box flex={{direction: 'row', gap: 4, alignItems: 'center'}}>
+            {metrics.pctChange && metrics.pctChange > 0 ? (
+              <Icon name="trending_up" size={16} color={Colors.textLighter()} />
+            ) : metrics.pctChange && metrics.pctChange < 0 ? (
+              <Icon name="trending_down" size={16} color={Colors.textLighter()} />
+            ) : null}
+            {percentFormatter.format(Math.abs(metrics.pctChange ?? 0))}
+          </Box>
+        </Box>
+      </Box>
+      <div className={styles.chartWrapper}>
+        <div className={styles.chartGraph}>
+          <Line
+            ref={chartRef}
+            data={getDataset(metrics, formatDatetime)}
+            options={options}
+            onClick={onClick}
+            updateMode="none"
+          />
+        </div>
+      </div>
+    </div>
+  );
+};
```

Comment:
Generic "line chart with comparison" component, replacing but heavily based on `AssetCatalogInsightsLineChart.tsx`, which is deleted in this PR.